src/mam4xx/drydep.hpp

```diff
@@ -154,8 +542,633 @@ Real gravit_settling_velocity(const Real particle_radius,
+    const bool lowest_model_layer,
+    const Real fraction_landuse[DryDeposition::n_land_type], const Real tair,
+    const Real pmid, const Real ram1, const Real fricvel,
+    const Real radius_part, const Real density_part, const Real sig_part,
+    const int moment, Real &vlc_dry, Real &vlc_trb, Real &vlc_grv) {
+
+  // use a maximum radius of 50 microns when calculating deposition velocity
+  static constexpr Real radius_max = 50.0e-6; //(BAD CONSTANT)
+
+  //------------------------------------------------------------------------------------
+  // Calculate deposition velocity of gravitational settling in all grid layers
+  //------------------------------------------------------------------------------------
+  vlc_grv = modal_aero_gravit_settling_velocity(
+      moment, radius_max, tair, pmid, radius_part, density_part, sig_part);
+
+  // vlc_dry is just the gravitational settling velocity for now.
+  vlc_dry = vlc_grv;
+
+  //------------------------------------------------------------------------------------
+  // For the lowest model layer:
+  //  - Calculate turbulent dry deposition velocity, vlc_trb.
+  //  - Add vlc_trb to vlc_grv to give the total deposition velocity, vlc_dry.
+  //------------------------------------------------------------------------------------
+  if (lowest_model_layer)
+    modal_aero_turb_drydep_velocity(moment, fraction_landuse, radius_max, tair,
+                                    pmid, radius_part, density_part, sig_part,
+                                    fricvel, ram1, vlc_grv, vlc_trb, vlc_dry);
+}
+
 } // namespace drydep
 
+// =============================================================================
+//  Main subroutine of aerosol dry deposition parameterization.
+//  Also serves as the interface routine called by EAM's physics driver.
+// =============================================================================
+KOKKOS_INLINE_FUNCTION
+void aero_model_drydep(
+    const ThreadTeam &team, const int nlev,
+    const Real fraction_landuse[DryDeposition::n_land_type],
+    const haero::ConstColumnView tair, const haero::ConstColumnView pmid,
+    const haero::ConstColumnView pint, const haero::ConstColumnView pdel,
+    const Diagnostics::ColumnTracerView state_q,
+    const ColumnView dgncur_awet[AeroConfig::num_modes()],
+    const ColumnView wetdens[AeroConfig::num_modes()],
+    const Kokkos::View<Real *> qqcw[ConvProc::gas_pcnst], const Real obklen,
+    const Real ustar, const Real landfrac, const Real icefrac,
+    const Real ocnfrac, const Real fricvelin, const Real ram1in,
+    const Diagnostics::ColumnTracerView ptend_q,
+    bool ptend_lq[ConvProc::gas_pcnst], const Real dt,
+    const ColumnView aerdepdrycw, const ColumnView aerdepdryis,
+
+    const ColumnView rho,
+    const Kokkos::View<Real *> vlc_dry[AeroConfig::num_modes()][4],
+    const Kokkos::View<Real *> vlc_trb[AeroConfig::num_modes()][4],
+    const Kokkos::View<Real *> vlc_grv[AeroConfig::num_modes()][4],
```

Comment:
It will be better to replace these `4`s with a static constexpr. Probably we should have done it in the Fortran codes as well.

Reply:
Agreed....